velox/docs/functions/spark/string.rst

```diff
@@ -59,6 +59,22 @@ Unless specified otherwise, all functions return NULL if at least one of the arg
 
         SELECT ltrim('ps', 'spark'); -- "ark"
 
+.. spark:function:: overlay(input, replace, pos, len) -> varchar
+
+    Replace a substring of ``input`` starting at ``pos`` character with ``replace`` and
+    going for rest ``len`` characters of ``input``.
+    First part is first pos - 1 characters of ``input``, ``pos`` is 1-based, if ``pos`` is less than or equal to 0,
+    it means abort first part of ``input``.
+    Second part is ``replace``.
+    Third part is rest of ``input`` from pos + len, if ``len`` is negative, it will be set to size of ``replace``,
```

Comment:
This is also confusing. Can len be zero?

```
If len is negative all of input starting at 'pos' and going to the end of input is replaced.
```

How can pos + len be negative? I assume pos is always >= 1 (if not, it is assumed to be 1) and len being negative just means go to the end of string. Am I missing something? Perhaps, add examples for all these edge cases.

Does this function support only VARCHAR or VARINARY as well? Please, clarify. If it supports VARBINARY, then I assume len is in bytes, not characters. Please, clarify as well.

Reply:
> Can len be zero?

Yes, consider this example `SELECT overlay('Spark SQL', 'ANSI ', 7, 0); -- "Spark ANSI SQL"`

> How can pos + len be negative?

`SELECT overlay('Spark SQL', '_', -6, 3); -- "_Sql"`

Reply:
I see. Let's clarify,

`When len is zero, inserts 'replace' at 'pos'.`

> SELECT overlay('Spark SQL', '_', -6, 3); -- "_Sql"

This is confusing to me. I'd expect this to replace "rk " with "_" and return "Spa_SQL". Can you explain again how this works?

Reply:
> SELECT overlay('Spark SQL', '_', -6, 3); -- "_Sql"

Assume we have a substring(string, start, len) function and start is one-based.
First parts is substring('Spark SQL', 1, -6), len is negative, so we get empty string.
Second parts is `replace`, so we get '_'.
Last parts is substring('Spark SQL', -6 + 3, INT_MAX), start is -3, it means count it from end and return last 3 characters, so we get 'SQL'.
Final results is `empty string` append `'_'` append `'SQL'`, `'_SQL'`

Hope it would make sense.

Reply:
> `When len is zero, inserts 'replace' at 'pos'.`

Actually, the place where 'replace' inserted has no related with 'len'